### PR TITLE
[Feature-7572][Server & ui] Add Backward Dependent Execution

### PR DESCRIPTION
**Pull request by shangeyao**
The downstream dependency is queried according to the dependent node, and the execution is triggered automatically.
#7572

### CODE REVIEW

**Comment by shangeyao** [comment]
At present, only "ALL" is considered.

File: dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessTaskRelationMapper.xml [hunk 1]
```diff
@@ -183,4 +183,14 @@
             and post_task_code = #{postTaskCode}
         </if>
     </select>
+    <select id="queryDistinctDefineCodeByTaskCode" resultType="org.apache.dolphinscheduler.dao.entity.ProcessTaskRelation">
+        select
+        process_definition_code
+        from t_ds_process_task_relation
+        WHERE pre_task_code = #{taskCode}
+        <if test="taskCode != 0">
+            or post_task_code = #{taskCode}
```
Comment (caishunfeng) [inline dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/ProcessTaskRelationMapper.xml:192]: I think it's better to split this sql instread of `or` query, which will dismatch index.

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java [hunk 2]
```diff
@@ -2716,4 +2721,33 @@ private void deleteCommandWithCheck(int commandId) {
             throw new ServiceException("delete command fail, id:" + commandId);
         }
     }
+
+    public List<ProcessDefinition> getPostDependentProcessDefinitionByCode(long processDefinitionCode) throws JsonProcessingException {
+        List<ProcessDefinition> result = new ArrayList<>();
+        List<TaskDefinition> definitionList = taskDefinitionLogMapper.queryDefinitionListByTaskType(TaskType.DEPENDENT.getDesc());
+        for (TaskDefinition taskDefinition : definitionList) {
+            String dependence = taskDefinition.getDependence();
+            ArrayNode dependTaskList = (ArrayNode) new ObjectMapper().readTree(dependence).get("dependTaskList");
```
Comment (caishunfeng) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java:2730]: It's better to use constant "dependTaskList"

File: dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java [hunk 3]
```diff
@@ -2716,4 +2721,33 @@ private void deleteCommandWithCheck(int commandId) {
...
+            for (JsonNode dependTask : dependTaskList) {
+                ArrayNode dependItemList = (ArrayNode) dependTask.get("dependItemList");
```
Comment (caishunfeng) [inline dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessService.java:2732]: It's better to use constant

**Comment by songjianet** [comment]
Thank you for your contribution, but it is currently in the switching period between the new and the old UI, and this function will not be implemented in the old UI.

**Pull request closed without merging**